### PR TITLE
feat: search component and browse rework

### PR DESCRIPTION
**Pull request by Gaisberg**
chore: improve nprogress coloring
refactor: use strong typing where logic implemented
chore: add shadcn-svelte components

### CODE REVIEW

File: src/lib/components/header.svelte [hunk 1]
```diff
@@ -119,3 +155,63 @@
 		</Drawer.Root>
 	</nav>
 </header>
+
+<Dialog.Root bind:open={isSearchOpen}>
+	<Dialog.Content class="p-0 sm:max-w-[600px] [&>button]:hidden">
```
Comment (AyushSehrawat) [inline src/lib/components/header.svelte:160]: `p-0 sm:max-w-[600px] [&>button]:hidden` See if removing `p-0 [&>button]:hidden` makes any difference. If not then remove them

File: src/lib/components/header.svelte [hunk 2]
```diff
@@ -1,44 +1,72 @@
...
+			searchResults = [];
+		}
+	}
+
+	async function handleMediaSearch() {
+		try {
+			const tmdbResults = await searchTMDB(searchQuery);
+			searchResults = [...tmdbResults];
+		} catch {
+			//pass
```
Comment (AyushSehrawat) [inline src/lib/components/header.svelte:54]: remove try catch or do catch (e) { console.error(e) }

**Review by AyushSehrawat — CHANGES_REQUESTED** [review]

File: src/routes/+layout.svelte [hunk 3]
```diff
@@ -8,26 +8,6 @@
...
-	// 	if (data.type === 'item_update') {
-	// 		const item = JSON.parse(data.item);
-	// 		toast.info(`${item.type} ${item.title} has been updated to ${item.state}`);
-	// 	}
-	// }
-
-	// const value = source('/api/sse').select('message');
-	// $: if ($value) {
-	// 	handleSSE($value);
-	// }
```
[the remark this answers is not in the document]
Comment (Gaisberg) [reply src/routes/+layout.svelte:30]: this logic is deprecated

File: src/lib/utils/search.ts
[the remark this answers is not in the document]
Comment (Gaisberg) [reply src/lib/utils/search.ts:1]: this is a future improvement

**Review by AyushSehrawat — APPROVED** [review]

**Pull request merged**